util/src/uri.cc

```diff
@@ -0,0 +1,75 @@
+#include <leatherman/util/uri.hpp>
+#include <algorithm>
+#include <sstream>
+
+namespace leatherman { namespace util {
+    uri::uri(std::string const& _uri)
+    {
+        if (_uri.length() == 0) {
+            return;
+        }
+
+        auto uri_end = _uri.end();
+
+        // get query start
+        auto query_start = std::find(_uri.begin(), uri_end, '?');
+
+        // protocol
+        auto protocol_start = _uri.begin();
+        auto protocol_end = std::find(protocol_start, uri_end, ':');
+
+        if (protocol_end != uri_end) {
+            std::string prot{protocol_end, uri_end};
```

Comment:
The variable name `prot` seems to imply this is capturing the protocol string, when in fact it's capturing everything *after* the protocol string.

Reply:
Updated.